RecoTracker/LST/plugins/LSTOutputConverter.cc

```diff
@@ -179,8 +179,8 @@ void LSTOutputConverter::produce(edm::StreamID, edm::Event& iEvent, const edm::E
       }
       if (lstTC_trackCandidateType[i] == LSTOutput::LSTTCType::T5)
         seed = seeds[0];
-      outputTS.emplace_back(seeds[0]);
-      auto const& ss = seeds[0].startingState();
+      outputTS.emplace_back(seed);
+      auto const& ss = seed.startingState();
```

Comment:
it looks like this file hasn't changed since May last year https://github.com/SegmentLinking/cmssw/commits/CMSSW_13_0_0_pre4_LST_X/RecoTracker/LST/plugins/LSTOutputConverter.cc
unfortunately, the history was reset in the move to 13_3_0_pre3.

I'm a bit surprised that this part has never failed before.

Back in https://github.com/SegmentLinking/cmssw/commit/3d19cab6e3f4cf77a9b79eedaaa6623fefa41f57#diff-5ea016ca574d1fc66969d2c8eb3ddddb002a3bc8ee03e0754d904f57e42fcb01L153-R152 the logic changed from making seeds on the fly for TCs without a seed (it was only T5s) to everything non-pLS (now T5, pT5, and pT3).

Another odd thing in that commit is now on L160 `      hitsForSeed.reserve(lstTC_len.size());` reserves the vector size for hits to be the size of the TC collection; but at least that's harmless.

The proposed change to use `seed` would go back to the pixel seeds to be used as initially intended, but then the code still filling the `seeds` vector becomes mostly useless and should probably change on L157 to be called just for `    if (lstTC_trackCandidateType[i] == LSTOutput::LSTTCType::T5) {` (and keep in mind that this means "the only type without a pixel seed", which is not the best way to maintain). Perhaps it's even better to go back to using `seedIdx != -1` in this part of the code. This will speed up this code quite a bit.

@VourMa was it important to make the seeds (`outputTS`) to contain the same hits as the full candidate (`outputTC`) for the HLT studies? 

Can we get a comparison for this PR by itself?
In the offline setup where the TCs go directly to refit, the content of `outputTS` is mostly irrelevant; so, I don't really expect any changes.

Reply:
> I'm a bit surprised that this part has never failed before.

There should definitely be a protection for an empty ```seeds``` but I am a bit surprised that we end up with nothing in that vector - is this an event that we have no pT5s, pT3s and T5s? How come this event didn't fail for the configuration with the higher pT threshold?

> Another odd thing in that commit is now on L160 ` hitsForSeed.reserve(lstTC_len.size());` reserves the vector size for hits to be the size of the TC collection; but at least that's harmless.

Indeed this is harmless but wrong. I guess the intention was for L160 to be:
```c++
hitsForSeed.reserve(lstTC_len[i]);
```

> Back in [3d19cab#diff-5ea016ca574d1fc66969d2c8eb3ddddb002a3bc8ee03e0754d904f57e42fcb01L153-R152](https://github.com/SegmentLinking/cmssw/commit/3d19cab6e3f4cf77a9b79eedaaa6623fefa41f57#diff-5ea016ca574d1fc66969d2c8eb3ddddb002a3bc8ee03e0754d904f57e42fcb01L153-R152) the logic changed from making seeds on the fly for TCs without a seed (it was only T5s) to everything non-pLS (now T5, pT5, and pT3).
> 
> The proposed change to use `seed` would go back to the pixel seeds to be used as initially intended, but then the code still filling the `seeds` vector becomes mostly useless and should probably change on L157 to be called just for ` if (lstTC_trackCandidateType[i] == LSTOutput::LSTTCType::T5) {` (and keep in mind that this means "the only type without a pixel seed", which is not the best way to maintain). Perhaps it's even better to go back to using `seedIdx != -1` in this part of the code. This will speed up this code quite a bit.
> 
> @VourMa was it important to make the seeds (`outputTS`) to contain the same hits as the full candidate (`outputTC`) for the HLT studies?
> 
> Can we get a comparison for this PR by itself? In the offline setup where the TCs go directly to refit, the content of `outputTS` is mostly irrelevant; so, I don't really expect any changes.

The logic was changed on purpose for experiment on the following questions: What happens if we pass extended seeds to mkFit - do we get extensions of the TrackCandidate and do we get the addition of more hits on the same layer? This is relevant for offline as well, unless the answer is obvious and we don't want to test it.
In the HLT context, only the pLS seeds were given as input to CKF:
https://github.com/SegmentLinking/cmssw/blob/bc0b5392c99cf51aefb2e8bd3a779bcbcba4000b/HLTrigger/Configuration/python/customizePhase2HLTTracking.py#L746
Still, I would probably not change the logic until we do the tests mentioned above. But we can discuss it and see what makes sense.

Reply:
> The logic was changed on purpose for experiment on the following questions: What happens if we pass extended seeds to mkFit - do we get extensions of the TrackCandidate and do we get the addition of more hits on the same layer?

OK, I had that afterthought after posting the message yesterday that the goal was to provide LST TCs as CMSSW `TrajectorySeed` and `TrackCandidate` vectors aiming to export the (almost) full list of hits in each TC.
In this case the proposed change in this PR is not correct (e.g. for a pT3 instead of a full IT+ OT hits it will pass along only the IT hits). A more appropriate choice would be to pick `seed` only if `seeds.empty()`.

I'm still interested to see why this candidate seed building fails: a printout of hit positions would help.

Beyond this PR, I think that we need to make the `TrajectorySeed`  (or `TrackCandidate`) collection when it's actually needed. Currently we are wasting CPU time to build a `TrajectorySeed`  collection and never use it.

Reply:
On your last point, what would be the proper way to do so? Is there a CMSSW functionality to run a specific part of the producer if a specific product is consumed downstream (I looked a bit in the twikis but I couldn't find something relevant)? Or are we talking about a brute force solution like adding a boolean in the `ParameterSet` of the producer to either make actual `TrajectorySeeds` or output a dummy collection without actually constructing them?

Reply:
> Or are we talking about a brute force solution like adding a boolean in the `ParameterSet` of the producer to either make actual `TrajectorySeeds` or output a dummy collection without actually constructing them?

I think that reconstruction/producers should be configured in the way needed to run a realistic job, not to make everything possibly doable.
So, yes, add a relevant flag to enable.